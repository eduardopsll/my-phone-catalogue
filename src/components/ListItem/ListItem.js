import React from "react";
import phoneIcon from "../../assets/icons/phone.svg";
import chevronIcon from "../../assets/icons/chevron.svg";

import styles from "./ListItem.module.scss";

function ListItem({ data, selectedItem, selectItemFn }) {
  const classSelectedItem = selectedItem === data.id ? styles["list-item--selected"] : "";
  return (
    <li
      className={`${styles["list-item"]} ${classSelectedItem}`}
      onClick={e => selectItemFn(data.id, e)}
    >
      <div className={styles["list-item__info-wrapper"]}>
        <img src={phoneIcon} className={styles["list-item__phone-icon"]} alt="phone-icon" />
        <div className={styles["list-item__text-wrapper"]}>
          {data.name}
          <span className={styles["list-item__text-subtitle"]}>{data.manufacturer}</span>
        </div>
        <div className={styles["list-item__price"]}>{data.price}â‚¬</div>
      </div>
      <img src={chevronIcon} className={styles["list-item__chevron-icon"]} alt="chevron-icon" />
    </li>
  );
}

export default ListItem;
