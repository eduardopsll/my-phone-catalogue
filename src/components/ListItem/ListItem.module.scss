@import "../../styles/base";

.list-item {
  align-items: center;
  border-bottom: 1px solid $secondary-color;
  color: $primary-color;
  cursor: pointer;
  display: flex;
  font-size: $font-size-m;
  font-weight: 700;
  justify-content: space-between;

  &:last-of-type {
    border: none;
  }

  &:hover {
    border-top: 2px solid $secondary-color;
    box-shadow: 0 6px 4px -4px rgba($primary-color, 0.75);
  }

  &__info-wrapper {
    align-items: center;
    display: flex;
    flex-grow: 1;
    padding: $space-3 $space-4;
  }

  &__phone-icon {
    height: pixel-to-rem(40);
    width: pixel-to-rem(40);
  }

  &__text-wrapper {
    display: flex;
    flex-direction: column;
    flex-grow: 1;
    padding-left: $space-4;
  }

  &__text-subtitle {
    color: $secondary-color;
    font-size: $font-size-s;
    font-weight: normal;
  }

  &__chevron-icon {
    flex-grow: 0;
    height: pixel-to-rem(25);
    width: pixel-to-rem(25);
  }
}
