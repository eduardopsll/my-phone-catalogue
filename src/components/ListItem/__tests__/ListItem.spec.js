import React from "react";
import ReactDOM from "react-dom";
import ReactTestUtils from "react-dom/test-utils";

import ListItem from "../ListItem";
import phonesMock from "../../../mocks/phones.json";

describe("ListItem", () => {
  it("should render without crashing", () => {
    const div = document.createElement("div");
    ReactDOM.render(<ListItem data={phonesMock[0]} />, div);
  });
  it("should call the selectItemFn when clicking over the list item", () => {
    const div = document.createElement("div");
    const selectItemFnMock = jest.fn();

    ReactDOM.render(<ListItem data={phonesMock[0]} selectItemFn={selectItemFnMock} />, div);

    const liNode = div.querySelector("li");
    ReactTestUtils.Simulate.click(liNode);
    expect(selectItemFnMock).toHaveBeenCalledTimes(1);
  });
  it("should add class selected when item is the selected one", () => {
    const div = document.createElement("div");
    const selectItemFnMock = jest.fn();

    ReactDOM.render(
      <ListItem data={phonesMock[0]} selectedItem={0} selectItemFn={selectItemFnMock} />,
      div
    );

    const liNode = div.querySelector("li");
    expect(liNode).toMatchSnapshot();
  });
  it("should not add class selected when there is no item selected", () => {
    const div = document.createElement("div");
    const selectItemFnMock = jest.fn();

    ReactDOM.render(
      <ListItem data={phonesMock[0]} selectedItem={null} selectItemFn={selectItemFnMock} />,
      div
    );

    const liNode = div.querySelector("li");
    expect(liNode).toMatchSnapshot();
  });
  it("should not add class selected when item is not the selected one", () => {
    const div = document.createElement("div");
    const selectItemFnMock = jest.fn();

    ReactDOM.render(
      <ListItem data={phonesMock[0]} selectedItem={null} selectItemFn={selectItemFnMock} />,
      div
    );

    const liNode = div.querySelector("li");
    expect(liNode).toMatchSnapshot();
  });
});
