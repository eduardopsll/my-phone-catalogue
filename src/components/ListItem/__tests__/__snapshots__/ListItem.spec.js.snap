// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ListItem should add class selected when item is the selected one 1`] = `
<li
  class="list-item list-item--selected"
>
  <div
    class="list-item__info-wrapper"
  >
    <img
      alt="phone-icon"
      class="list-item__phone-icon"
      src="phone.svg"
    />
    <div
      class="list-item__text-wrapper"
    >
      iPhone 7
      <span
        class="list-item__text-subtitle"
      >
        Apple
      </span>
    </div>
    <div
      class="list-item__price"
    >
      769
      €
    </div>
  </div>
  <img
    alt="chevron-icon"
    class="list-item__chevron-icon"
    src="chevron.svg"
  />
</li>
`;

exports[`ListItem should not add class selected when item is not the selected one 1`] = `
<li
  class="list-item "
>
  <div
    class="list-item__info-wrapper"
  >
    <img
      alt="phone-icon"
      class="list-item__phone-icon"
      src="phone.svg"
    />
    <div
      class="list-item__text-wrapper"
    >
      iPhone 7
      <span
        class="list-item__text-subtitle"
      >
        Apple
      </span>
    </div>
    <div
      class="list-item__price"
    >
      769
      €
    </div>
  </div>
  <img
    alt="chevron-icon"
    class="list-item__chevron-icon"
    src="chevron.svg"
  />
</li>
`;

exports[`ListItem should not add class selected when there is no item selected 1`] = `
<li
  class="list-item "
>
  <div
    class="list-item__info-wrapper"
  >
    <img
      alt="phone-icon"
      class="list-item__phone-icon"
      src="phone.svg"
    />
    <div
      class="list-item__text-wrapper"
    >
      iPhone 7
      <span
        class="list-item__text-subtitle"
      >
        Apple
      </span>
    </div>
    <div
      class="list-item__price"
    >
      769
      €
    </div>
  </div>
  <img
    alt="chevron-icon"
    class="list-item__chevron-icon"
    src="chevron.svg"
  />
</li>
`;
